### ISSUE TITLE
Error: Unbound module Frenetic_OpenFlow0x01_Plugin

### ISSUE THREAD

**Issue by rodrigosoares** [merlin-lang/merlin#1]
Hi.

I am trying to install Merlin following the instructions as stated in the README.md file, but I am running into the issue below:

```
File "lib/Merlin_Controller.ml", line 11, characters 12-49:
Error: Unbound module Frenetic_OpenFlow0x01_Plugin
Command exited with code 2.
Compilation unsuccessful after building 72 targets (0 cached) in 00:00:15.
E: Failure("Command ''/usr/bin/ocamlbuild' lib/libmerlin_stubs.a lib/dllmerlin_stubs.so lib/merlin.cma lib/merlin.cmxa lib/merlin.a lib/merlin.cmxs lib/Merlin.native test/Test.native -use-ocamlfind -tag debug' terminated with error code 10")
make: *** [build] Error 1
```

This error occurs when I run `make` inside Merlin directory. It seems to be something related to missing libraries. I have installed all the necessary packages and I downloaded Merlin in the home directory of the Frenetic VM available in its [Git repository](https://github.com/frenetic-lang/frenetic), since there is no instructions about where to put the code in the VM file system.

I also tried to update Frenetic code from the VM to a more recent version, since it has a `Frenetic_OpenFlow0x01_Plugin.ml` module, but this leads to version incompatibilities with other dependencies.

**Comment by jnfoster** [comment]
Hi Rodrigo,

Merlin (master) compiles just fine for me against Frenetic (master). I
don't know what VM you're using or what it's current state is.

Can you provide more details about precisely which packages (and which
versions) you have installed? I'm afraid this

I also tried to update Frenetic code from the VM to a more recent version,
> since it has a Frenetic_OpenFlow0x01_Plugin.ml module, but this leads to
> version incompatibilities with other dependencies.

is too vague for me to be able to help you.

-N

**Comment by rodrigosoares** [comment]
Hi, Nate,

The VM I am using is the one available at Frenetic's repo, but I've found a solution for my problem already. I'll leave here all the steps I took to solve this in case someone else runs into the same problem.

The VM Frenetic team made available does not have the Frenetic code at its current master version. This "Frenetic_OpenFlow0x01_Plugin.ml" problem I was facing was because I was trying to run Merlin (master) against Frenetic version in the VM, which was not master (although they say so). In the current Frenetic master version, this file "Frenetic_OpenFlow0x01_Plugin.ml" was renamed to "Frenetic_OpenFlow0x01.ml", besides other changes.

Basically, I had to do the following steps:
1. `git pull`'ed the installed Frenetic version in the VM to current master
2. Installed OCaml compiler version 4.03.0 with `opam switch 4.03.0`
3. Installed all dependecies of Frenetic and Merlin for OCaml 4.03.0
4. Installed "dprle" package version 0.6.0 from Git repo forked by Frenetic team
5. Moved directory `~/.opam/system/lib/dprle` to `~/.opam/4.03.0/lib/dprle`
6. Built Merlin normally as instructed in README.

Anyway, thanks for the answer. Best regards!
